src/devices_models/devices/common/objective_functions.jl

```diff
@@ -525,15 +525,15 @@ function _add_pwl_term!(
         # First element of the return is the average cost at P_min.
         # Shouldn't be passed for convexity check
         is_convex = _slope_convexity_check(slopes[2:end])
-        break_points = PSY.get_breakpoint_upperbounds(data) ./ base_power
+        break_points = map(x -> last(x), data) ./ base_power
+        if has_container_key(container, OnStatusParameter, T)
```

Comment:
This block of code container a relatively complex logic, we should abstract it into a function.